models/RegistrationChecks.php

```diff
@@ -135,11 +135,28 @@ public function showAgeCheck()
 
         /** @var Module $module */
         $module = Yii::$app->getModule('legal');
-        if ($module->showAgeCheck()) {
-            if ($this->user === null || empty($module->settings->user($this->user)->get(static::SETTING_KEY_AGE))) {
+
+        // Check if the user and their profile property are not null before accessing them
+        if ($module->showAgeCheck() && $this->user !== null && $this->user->profile !== null) {
+            // Calculate the user's age
+            $birthday = $this->user->profile->birthday;
```

Comment:
Maybe we could move this code into a seperate private method?

Reply:
We can attempt do to this if required, would you want the call to also be made within this function through a static method or not?

Reply:
I don't think a static method will be necessary.

Reply:
Looking closer, we'll also have to do something within this class as well else it doesn't really do anything for `EVENT_BEFORE_VALIDATE` as the current https://github.com/humhub-contrib/legal/pull/67#issuecomment-1962238125 throws an error calling for `Profile::models` which doesn't exist. 🤔